tests/dispatch/test_http2.py

```diff
@@ -1,51 +1,108 @@
+import asyncio
 import json
+import sys
+import threading
 
 import h2.connection
 import h2.events
+import pytest
 from h2.settings import SettingCodes
+from hypercorn.asyncio import serve
+from hypercorn.config import Config
 
 from httpx import AsyncClient, Client, Response
 
 from .utils import MockHTTP2Backend
 
 
-def app(request):
-    content = json.dumps(
-        {
-            "method": request.method,
-            "path": request.url.path,
-            "body": request.content.decode(),
-        }
-    ).encode()
-    headers = {"Content-Length": str(len(content))}
-    return Response(200, headers=headers, content=content)
-
-
-def test_http2_get_request():
-    backend = MockHTTP2Backend(app=app)
-
-    with Client(backend=backend) as client:
-        response = client.get("http://example.org")
+async def app(scope, receive, send):
+    if scope["type"] == "lifespan":
+        while True:
+            message = await receive()
+            if message["type"] == "lifespan.startup":
+                await send({"type": "lifespan.startup.complete"})
+            elif message["type"] == "lifespan.shutdown":
+                await send({"type": "lifespan.shutdown.complete"})
+                return
+    elif scope["type"] == "http":
+        assert scope["http_version"] == "2"
+
+        body = b""
+        more_body = True
+        while more_body:
+            message = await receive()
+            body += message.get("body", b"")
+            more_body = message.get("more_body", False)
+
+        content = json.dumps(
+            {"method": scope["method"], "path": scope["path"], "body": body.decode()}
+        ).encode()
+        headers = [(b"content-length", b"%d" % len(content))]
+        await send({"type": "http.response.start", "status": 200, "headers": headers})
+        await send({"type": "http.response.body", "body": content, "more_body": False})
+    else:
+        raise Exception()
+
+
+@pytest.fixture(name="h2_server", scope="module")
+def _h2_server(cert_pem_file, cert_private_key_file):
+    bind = "0.0.0.0:8543"
```

Comment:
This is one problem - whilst Hypercorn can bind to a random available port (0.0.0.0:0) there isn't an easy way (unless you use Trio) to figure out which port it has bound to. Actually I think I can just switch this to Trio, right?